src/OSPSuite.XMLWrapper/src/XMLHelper.cpp

```diff
@@ -14,297 +15,295 @@
+   {
+      pXMLDoc.Release();
+      throw ErrorData(ErrorData::ED_ERROR, "XMLHelper::GetModelNodeFromFile", "Failed to find Model Node in XML string");
+   }
+
+   XMLNode modelNodeCopy = pModelNode.Clone(true);
+   pXMLDoc.Release();
+   return modelNodeCopy;
 }
 
-std::string XMLHelper::GetParseError (XMLParseErrorPtr pError)
+std::string XMLHelper::GetParseError(XMLParseErrorPtr pError)
 {
 #ifdef _WINDOWS
-	std::string sErr;
-	sErr = "Error code: " + _bstr_t(pError->GeterrorCode()) + "\n";
-	try
-	{ //This calls might failed
-		sErr += std::string("Url: ") + BStrToSTDString(pError->Geturl()) + "\n";
-		sErr += std::string("Source Text: ") + BStrToSTDString(pError->GetsrcText()) + "\n";
-    }
-    catch(...)
-    {
-    	sErr = sErr;
-    }
-    sErr += "At line: " +  _bstr_t(pError->Getline()) + "\n";
-    sErr += "At line Pos: " +  _bstr_t(pError->Getlinepos()) + "\n";
-	sErr += std::string("Reason: ") + BStrToSTDString(pError->Getreason());
-	return sErr;
+   std::string sErr;
+   sErr = "Error code: " + _bstr_t(pError->GeterrorCode()) + "\n";
+   try
+   { //This calls might failed
+      sErr += std::string("Url: ") + BStrToSTDString(pError->Geturl()) + "\n";
+      sErr += std::string("Source Text: ") + BStrToSTDString(pError->GetsrcText()) + "\n";
+   }
+   catch (...)
+   {
+      sErr = sErr;
+   }
+   sErr += "At line: " + _bstr_t(pError->Getline()) + "\n";
+   sErr += "At line Pos: " + _bstr_t(pError->Getlinepos()) + "\n";
+   sErr += std::string("Reason: ") + BStrToSTDString(pError->Getreason());
+   return sErr;
 #endif
 
 #ifdef linux
-	// !!! Not implemented yet
-	return "Unspecified error (XMLHelper::GetParseError() not implemented yet for Linux)";
+   // !!! Not implemented yet
+   return "Unspecified error (XMLHelper::GetParseError() not implemented yet for Linux)";
 #endif
 }
 
-double XMLHelper::ToDouble (const std::string & sDouble)
+double XMLHelper::ToDouble(const std::string& sDouble)
 {
-	// Return value
-	std::string s = Trim(sDouble);
+   std::string s = Trim(sDouble);
 
-	if (!IsNumeric(s))
-		throw ErrorData(ErrorData::ED_ERROR, "XMLHelper::ToDouble", "Not Numeric:" + sDouble);
+   if (!IsNumeric(s))
+      throw ErrorData(ErrorData::ED_ERROR, "XMLHelper::ToDouble", "Not Numeric:" + sDouble);
 
-    const char * DecSep = localeconv()->decimal_point;
+   const char* DecSep = localeconv()->decimal_point;
 
-    if (strcmp(",", DecSep))
-        s = XMLHelper::StringReplace(s, ",", DecSep);
+   if (strcmp(",", DecSep))
+      s = XMLHelper::StringReplace(s, ",", DecSep);
 
-    if (strcmp(".", DecSep))
-        s = XMLHelper::StringReplace(s, ".", DecSep);
+   if (strcmp(".", DecSep))
+      s = XMLHelper::StringReplace(s, ".", DecSep);
 
-	double value = 0.;
+   double value = 0.;
+   char* end;
 
-	// String stream
-	std::istringstream sink(s);
+   value = strtod(s.c_str(), &end);
 
-	// Read value from string stream
-	sink >> value;
+   if (*end != '\0')
+      return SimModelNative::MathHelper::GetNaN();
```

Comment:
This is the change. The old code (below) worked well when converting a finite number within a string to double.
But if the string was "infinity", "NaN", etc - it converted it to zero.
```
std::istringstream sink(s);
sink >> value;
```